Anagram.java

```diff
@@ -2,48 +2,85 @@
 public class Anagram {
 	public static void main(String args[]) {
 		// Tests the isAnagram function.
-		System.out.println(isAnagram("silent","listen"));  // true
-		System.out.println(isAnagram("William Shakespeare","I am a weakish speller")); // true
-		System.out.println(isAnagram("Madam Curie","Radium came")); // true
-		System.out.println(isAnagram("Tom Marvolo Riddle","I am Lord Voldemort")); // true
+		System.out.println(isAnagram("silent", "listen")); // true
+		System.out.println(isAnagram("William Shakespeare", "I am a weakish speller")); // true
+		System.out.println(isAnagram("Madam Curie", "Radium came")); // true
+		System.out.println(isAnagram("Tom Marvolo Riddle", "I am Lord Voldemort")); // true
 
 		// Tests the preProcess function.
 		System.out.println(preProcess("What? No way!!!"));
-		
+
 		// Tests the randomAnagram function.
 		System.out.println("silent and " + randomAnagram("silent") + " are anagrams.");
-		
-		// Performs a stress test of randomAnagram 
+
+		// Performs a stress test of randomAnagram
 		String str = "1234567";
 		Boolean pass = true;
 		//// 10 can be changed to much larger values, like 1000
 		for (int i = 0; i < 10; i++) {
 			String randomAnagram = randomAnagram(str);
 			System.out.println(randomAnagram);
 			pass = pass && isAnagram(str, randomAnagram);
-			if (!pass) break;
+			if (!pass)
+				break;
 		}
 		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+	}
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		if (str1.isEmpty() && str2.isEmpty()) {
+			return true;
+		}
+		boolean anagram;
```

Comment:
dont declare variable without initial value. 
`string.toLowerCase()` wasn't taught in recitation or lecture, therefore it is hw guidelines violation. -2; 
Alternative: Implement yourself.